test/lifecycle_module_test.dart

```diff
@@ -1855,4 +2336,31 @@ void main() {
       expect(result.messagesAsString(), equals('mock message\nmock message 2'));
     });
   });
+
+  if (runSpanTests) {
+    group('with an unnamed module', () {
+      UnnamedModule module;
+      setUp(() {
+        module = new UnnamedModule();
+      });
+
+      tearDown(() async {
+        await module.dispose();
+      });
+
+      test('does not create spans for transitions', () async {
+        final sub = getTestTracer().onSpanFinish.listen((_) {
+          fail('No span should be created');
+        });
+        addTearDown(sub.cancel);
+
+        await module.load();
+        await module.suspend();
+        await module.resume();
+        await module.unload();
+        // Wait some time after transitions are over to be sure no spans are created
+        await new Future.delayed(new Duration(milliseconds: 10));
+      });
```

Comment:
Sorry, saw one more thing while reviewing your dependent PR...

Can we also add a test for an unnamed module that has loaded a child module, to assert all the null-checks on the `_activeSpan` are working as intended?

Reply:
Added